ui/helpers/utils/permission.js

```diff
@@ -245,88 +244,68 @@ export const PERMISSION_DESCRIPTIONS = deepFreeze({
         getSlip44ProtocolName(coinType) ??
         `${t('unknownNetworkForKeyEntropy')} m/44'/${coinType}'`,
     })),
-  [RestrictedMethods.snap_getEntropy]: ({ t, targetSubjectMetadata }) => ({
-    label: t('permission_getEntropy', [
-      getSnapNameComponent(targetSubjectMetadata),
-    ]),
+  [RestrictedMethods.snap_getEntropy]: ({ t, snapName }) => ({
+    label: t('permission_getEntropy', [getSnapNameComponent(snapName)]),
     description: t('permission_getEntropyDescription', [
-      getSnapNameComponent(targetSubjectMetadata),
+      getSnapNameComponent(snapName),
     ]),
     leftIcon: IconName.SecurityKey,
     weight: 4,
   }),
 
-  [RestrictedMethods.snap_manageState]: ({ t, targetSubjectMetadata }) => ({
+  [RestrictedMethods.snap_manageState]: ({ t, snapName }) => ({
     label: t('permission_manageState'),
     description: t('permission_manageStateDescription', [
-      getSnapNameComponent(targetSubjectMetadata),
+      getSnapNameComponent(snapName),
     ]),
     leftIcon: IconName.AddSquare,
     weight: 4,
   }),
-  [RestrictedMethods.snap_getLocale]: ({ t, targetSubjectMetadata }) => ({
+  [RestrictedMethods.snap_getLocale]: ({ t, snapName }) => ({
     label: t('permission_getLocale'),
     description: t('permission_getLocaleDescription', [
-      getSnapNameComponent(targetSubjectMetadata),
+      getSnapNameComponent(snapName),
     ]),
     leftIcon: IconName.Global,
     weight: 4,
   }),
-  [RestrictedMethods.wallet_snap]: ({
-    t,
-    permissionValue,
-    targetSubjectMetadata,
-  }) => {
+  [RestrictedMethods.wallet_snap]: ({ t, permissionValue, snapName }) => {
     const snaps = permissionValue.caveats[0].value;
     const baseDescription = {
       leftIcon: getLeftIcon(IconName.Flash),
       rightIcon: RIGHT_INFO_ICON,
     };
 
     return Object.keys(snaps).map((snapId) => {
-      const friendlyName = getSnapName(snapId, targetSubjectMetadata);
-      if (friendlyName) {
-        return {
-          ...baseDescription,
-          label: t('permission_accessNamedSnap', [
-            <Text
-              color={TextColor.inherit}
-              variant={TextVariant.inherit}
-              fontWeight={FontWeight.Medium}
-              key={snapId}
-            >
-              {friendlyName}
-            </Text>,
-          ]),
-          description: t('permission_accessSnapDescription', [friendlyName]),
-        };
-      }
-
+      const friendlyName = snapName;
```

Comment:
This is flawed. The name here should be the name of the requested snap as part of the permission, not the installing snap. Also we may not have a name for this snap, so a bunch of code needs to be reverted here